sql-plugin/src/main/scala/com/nvidia/spark/rapids/implicits.scala

```diff
@@ -87,11 +111,44 @@ object RapidsPluginImplicits {
     }
   }
 
+  implicit class RapidsBufferSeq[A <: RapidsBuffer](val in: SeqLike[A, _]) {
+    /**
+     * safeFree: Is an implicit on a sequence of RapidsBuffer classes that tries to free each
+     * element of the sequence, even if prior free calls fail. In case of failure in any of the
+     * free calls, an Exception is thrown containing the suppressed exceptions (getSuppressed),
+     * if any.
+     */
+    def safeFree(error: Throwable = null): Unit = if (in != null) {
```

Comment:
looks like significant duplication between safeFree and safeCLose? Can you try to extract common logic into a common base class, and maybe use a lambda `fn: A => Unit` to implement the difference between close and free?